src/components/ProductList.jsx

```diff
@@ -38,60 +41,105 @@ export default function ProductList({ products }) {
         }),
       }).then(async (response) => {
         let data = await response.json();
-        alert(data.message)
+        alert(data.message);
         console.log(response.status);
       });
     } catch (error) {
       console.error('error: ', error);
     }
-  }
+  };
+
+  const getProduct = async (productId) => {
+    const url = process.env.NEXT_PUBLIC_DOMAIN;
+    const res = await fetch(
+      `${url}/api/product-compare?productId=${productId}`,
```

Comment:
The modal was stuck in the loading phase when I ran the app. I did not know that I needed to add the `NEXT_PUBLIC_DOMAIN` variable in my .env file, but a good solution to this since you are calling fetch in the client-side would be to just use the `../api/product-compare?productId=${productId}` instead of `${url}/api/product-compare?productId=${productId}`. Hope this helps but overall looks good to me 👍